pandas/io/excel.py

```diff
@@ -296,76 +316,96 @@ def _parse_excel(self, sheetname=0, header=0, skiprows=None, skip_footer=0,
-
-        data = []
-        should_parse = {}
-        for i in range(sheet.nrows):
-            row = []
-            for j, (value, typ) in enumerate(zip(sheet.row_values(i),
-                                                 sheet.row_types(i))):
-                if parse_cols is not None and j not in should_parse:
-                    should_parse[j] = self._should_parse(j, parse_cols)
-
-                if parse_cols is None or should_parse[j]:
-                    if typ == XL_CELL_DATE:
-                        if xlrd_0_9_3:
-                            # Use the newer xlrd datetime handling.
-                            value = xldate.xldate_as_datetime(value, epoch1904)
-
-                            # Excel doesn't distinguish between dates and time,
-                            # so we treat dates on the epoch as times only.
-                            # Also, Excel supports 1900 and 1904 epochs.
-                            year = (value.timetuple())[0:3]
-                            if ((not epoch1904 and year == (1899, 12, 31))
-                                    or (epoch1904 and year == (1904, 1, 1))):
-                                value = datetime.time(value.hour,
-                                                      value.minute,
-                                                      value.second,
-                                                      value.microsecond)
-                        else:
-                            # Use the xlrd <= 0.9.2 date handling.
-                            dt = xldate.xldate_as_tuple(value, epoch1904)
-
-                            if dt[0] < datetime.MINYEAR:
-                                value = datetime.time(*dt[3:])
+        
+        #Keep sheetname to maintain backwards compatibility.
+        if isinstance(sheetname, list):
+            sheets = sheetname
+        elif sheetname is None:
+            sheets = self.sheet_names
+        else:
+            sheets = [sheetname]
+        
+        output = {}
+        
+        for asheetname in sheets:
+            if verbose:
+                print("Reading sheet %s" % asheetname)
+            
+            if isinstance(asheetname, compat.string_types):
+                sheet = self.book.sheet_by_name(asheetname)
+            else:  # assume an integer if not a string
+                sheet = self.book.sheet_by_index(asheetname)
+            
+            data = []
+            should_parse = {}
+            for i in range(sheet.nrows):
+                row = []
+                for j, (value, typ) in enumerate(zip(sheet.row_values(i),
+                                                     sheet.row_types(i))):
+                    if parse_cols is not None and j not in should_parse:
+                        should_parse[j] = self._should_parse(j, parse_cols)
+    
+                    if parse_cols is None or should_parse[j]:
+                        if typ == XL_CELL_DATE:
```

Comment:
I would put all of this in to a locally defined function to make this code a bit more clean, e.g. something like ``parse_cell`` (you can define above)